api/src/org/apache/cloudstack/api/command/admin/outofbandmanagement/IssueOutOfBandManagementPowerActionCmd.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.cloudstack.api.response.OutOfBandManagementResponse;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.cloudstack.outofbandmanagement.OutOfBandManagement.PowerOperation;
+import org.apache.cloudstack.outofbandmanagement.OutOfBandManagementService;
+
+import javax.inject.Inject;
+
+@APICommand(name = "issueOutOfBandManagementPowerAction", description = "Initiates the specified power action to the host's out-of-band management interface",
+        responseObject = OutOfBandManagementResponse.class, requestHasSensitiveInfo = false, responseHasSensitiveInfo = false, authorized = {RoleType.Admin})
+public class IssueOutOfBandManagementPowerActionCmd extends BaseAsyncCmd {
+    @Inject
+    private OutOfBandManagementService outOfBandManagementService;
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.HOST_ID, type = CommandType.UUID, entityType = HostResponse.class, required = true, description = "the ID of the host")
+    private Long hostId;
+
+    @Parameter(name = ApiConstants.TIMEOUT, type = CommandType.LONG, description = "optional operation timeout in seconds that overrides the global or cluster-level out-of-band management timeout setting")
+    private Long actionTimeout;
+
+    @Parameter(name = ApiConstants.ACTION, type = CommandType.STRING, required = true, description = "out-of-band management power actions, valid actions are: ON, OFF, CYCLE, RESET, SOFT, STATUS")
+    private String outOfBandManagementPowerOperation;
+
+    /////////////////////////////////////////////////////
+    /////////////// API Implementation///////////////////
+    /////////////////////////////////////////////////////
+
+    @Override
+    public String getCommandName() {
+        return "issueoutofbandmanagementpoweractionresponse";
+    }
+
+    private void validateParams() {
+        if (getHostId() == null || getHostId() < 1L) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Invalid host ID: " + getHostId());
+        }
+        if (Strings.isNullOrEmpty(getOutOfBandManagementPowerOperation())) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Invalid out-of-band management power action: " + getOutOfBandManagementPowerOperation());
+        }
+    }
+
+    @Override
+    public void execute() throws ResourceUnavailableException, InsufficientCapacityException, ServerApiException, ConcurrentOperationException, ResourceAllocationException, NetworkRuleConflictException {
+        validateParams();
+        Host host = _resourceService.getHost(getHostId());
+        if (host == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Unable to find host by ID: " + getHostId());
+        }
+        PowerOperation powerOperation = PowerOperation.valueOf(getOutOfBandManagementPowerOperation());
```

Comment:
@rhtyd the API itself is case-insensitive.  Therefore, I am more concerned that the user will received an unexpected error because we are not fulfilling the contract.  For this reason, it seems like we should be doing a case intensive check here and in ``validateParams``.